src/cli/serve/integration_tests/invalid_config.test.ts

```diff
@@ -33,18 +32,25 @@ describe.skip('cli invalid config support', function () {
         }
       );
 
-      const [fatalLogLine] = stdout
-        .toString('utf8')
-        .split('\n')
-        .filter(Boolean)
-        .map((line) => JSON.parse(line) as LogEntry)
-        .filter((line) => line.tags.includes('fatal'))
-        .map((obj) => ({
-          ...obj,
-          pid: '## PID ##',
-          '@timestamp': '## @timestamp ##',
-          error: '## Error with stack trace ##',
-        }));
+      let fatalLogLine;
+      try {
+        [fatalLogLine] = stdout
+          .toString('utf8')
+          .split('\n')
+          .filter(Boolean)
+          .map((line) => JSON.parse(line) as LogEntry)
+          .filter((line) => line.tags.includes('fatal'))
```

Comment:
The test will fail anyway due to the cause, but this time it'll fail with the new error we are throwing.

The test was parsing these entries:
```
{"log.level":"error","@timestamp":"2021-04-29T04:56:27.104Z","log":{"logger":"elastic-apm-node"},"ecs":{"version":"1.6.0"},"message":"metadata requests did not finish, possible deadlock"}
{"type":"log","@timestamp":"2021-04-29T04:57:47+00:00","tags":["info","plugins-service"],"pid":4478,"message":"Plugin \"timelines\" is disabled."}
{"type":"log","@timestamp":"2021-04-29T04:57:47+00:00","tags":["warning","config","deprecation"],"pid":4478,"message":"plugins.scanDirs is deprecated and is no longer used"}
{"type":"log","@timestamp":"2021-04-29T04:57:47+00:00","tags":["warning","config","deprecation"],"pid":4478,"message":"\"xpack.reporting.roles\" is deprecated. Granting reporting privilege through a \"reporting_user\" role will not be supported starting in 8.0. Please set 'xpack.reporting.roles.enabled' to 'false' and grant reporting privilege to users through feature controls in Management > Security > Roles"}
{"type":"log","@timestamp":"2021-04-29T04:57:47+00:00","tags":["fatal","root"],"pid":4478,"message":"CriticalError: Unknown configuration key(s): \"unknown.key\", \"other.unknown.key\", \"other.third\", \"some.flat.key\", \"some.array\". Check for spelling errors and ensure that expected plugins are installed.\n    at ensureValidConfiguration (/dev/shm/workspace/parallel/12/kibana/src/core/server/config/ensure_valid_configuration.ts:35:11)\n    at Server.setup (/dev/shm/workspace/parallel/12/kibana/src/core/server/server.ts:179:5)\n    at Root.setup (/dev/shm/workspace/parallel/12/kibana/src/core/server/root/index.ts:44:14)\n    at bootstrap (/dev/shm/workspace/parallel/12/kibana/src/core/server/bootstrap.ts:97:5)\n    at Command.<anonymous> (/dev/shm/workspace/parallel/12/kibana/src/cli/serve/serve.js:224:5) {\n  code: 'InvalidConfig',\n  processExitCode: 64,\n  cause: undefined\n}"}
```

The first line:
```
{"log.level":"error","@timestamp":"2021-04-29T04:56:27.104Z","log":{"logger":"elastic-apm-node"},"ecs":{"version":"1.6.0"},"message":"metadata requests did not finish, possible deadlock"}
```

It does not include the `tags` field because it's logged by an external module: `elastic-apm-node`.

I think we need to optionally access to the tags field:
```suggestion
          .filter((line) => line.tags?.includes('fatal'))
```

And probably change the `LogEntry` interface declaration to set `tags` and `type` as optional.

Reply:
Yeah, honestly, wasn't intending on fixing the issue with this code but then the test proved to be failing a LOT more than expected